tasks/postgresql.yml

```diff
@@ -0,0 +1,162 @@
+---
+- name: "create secret for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: Secret
+      apiVersion: v1
+      metadata:
+        annotations:
+          template.openshift.io/expose-database_name: "{.data['database-name']}"
+          template.openshift.io/expose-password: "{.data['database-password']}"
+          template.openshift.io/expose-username: "{.data['database-user']}"
+        name: "{{ DATABASE_SERVICE_NAME }}"
+        namespace: "{{ NAMESPACE }}"
+      stringData:
+        database-name: "{{ POSTGRESQL_DATABASE }}"
+        database-password: "{{ POSTGRESQL_PASSWORD }}"
+        database-user: "{{ POSTGRESQL_USER }}"
+
+- name: "create service for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: Service
+      apiVersion: v1
+      metadata:
+        annotations:
+          template.openshift.io/expose-uri: postgres://{.spec.clusterIP}:{.spec.ports[?(.name=="postgresql")].port}
+        name: "{{ DATABASE_SERVICE_NAME }}"
+        namespace: "{{ NAMESPACE }}"
+      spec:
+        ports:
+        - name: postgresql
+          nodePort: 0
+          port: 5432
+          protocol: TCP
+          targetPort: 5432
+        selector:
+          name: "{{ DATABASE_SERVICE_NAME }}"
+        sessionAffinity: None
+        type: ClusterIP
+      status:
+        loadBalancer: {}
+
+- name: "create presistant volume claim for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: PersistentVolumeClaim
+      apiVersion: v1
+      metadata:
+        name: "{{ DATABASE_SERVICE_NAME }}"
+        namespace: "{{ NAMESPACE }}"
+      spec:
+        accessModes:
+        - ReadWriteOnce
+        resources:
+          requests:
+            storage: "{{ VOLUME_CAPACITY }}"
+        storageClassName: "{{ STORAGE_CLASS_NAME }}"
+
+- name: "create secret for ceph based persistent volume"
+  k8s:
+    state: present
+    definition:
+      kind: Secret
+      apiVersion: v1
+      type: kubernetes.io/rbd
+      metadata:
```

Comment:
Label `app=thoth`.

Reply:
if we are gonna keep the pvc by not adding the label to the pvc, then we shouldn't add the label to pvc secret as well. WDYT?

Reply:
Makes sense :+1: